recipes/default.rb

```diff
@@ -21,42 +21,40 @@
   package_action node['openldap']['package_install_action']
 end
 
-case node['platform_family']
-when 'debian'
-  template '/etc/default/slapd' do
-    source 'default_slapd.erb'
-    notifies :restart, 'service[slapd]'
-  end
-when 'rhel', 'amazon'
-  template '/etc/sysconfig/slapd' do
-    source 'sysconfig_slapd.erb'
-    notifies :restart, 'service[slapd]'
-  end
-when 'suse'
-  template '/etc/sysconfig/openldap' do
-    source 'sysconfig_openldap.erb'
-    notifies :restart, 'service[slapd]'
-  end
-when 'freebsd'
-  template '/etc/rc.conf.d/slapd' do
-    source 'rc_slapd.erb'
-    notifies :restart, 'service[slapd]'
-  end
+template openldap_defaults_path do
+  source openldap_defaults_template
+  notifies :restart, 'service[slapd]'
 end
 
 ##  Set syncrepl_consumer_config dynamic values here
 node.default_unless['openldap']['syncrepl_consumer_config']['searchbase'] = "\"#{node['openldap']['basedn']}\""
 node.default_unless['openldap']['syncrepl_consumer_config']['binddn'] = "\"#{node['openldap']['syncrepl_cn']},#{node['openldap']['basedn']}\""
 node.default_unless['openldap']['syncrepl_consumer_config']['credentials'] = "\"#{node['openldap']['slapd_replpw']}\""
 
-template "#{node['openldap']['dir']}/slapd.conf" do
+systemd_unit 'slapd.service' do
+  content openldap_el8_systemd_unit
+  action [:create]
+end if openldap_el8_systemd_unit?
```

Comment:
Should this (& others) be `only_if` guards instead?

Reply:
Doing it this way will allow the resource not to show up in the chef runs (even as not doing anything). I believe this has been an standard way of doing things in Sous Chefs as of late but I might be wrong.